src/openrct2/GameState.h

```diff
@@ -11,12 +11,54 @@
 
 #include "Date.h"
 
+#include <array>
+#include <chrono>
+#include <map>
 #include <memory>
 
 namespace OpenRCT2
 {
     class Park;
 
+    // Information regarding various pieces of logic update
+    enum class LogicTimePart
+    {
+        NetworkUpdate,
+        Date,
+        Scenario,
+        Climate,
+        MapTiles,
+        MapStashProvisionalElements,
+        MapPathWideFlags,
+        Peep,
+        MapRestoreProvisionalElements,
+        Vehicle,
+        Misc,
+        Ride,
+        Park,
+        Research,
+        RideRatings,
+        RideMeasurments,
+        News,
+        MapAnimation,
+        Sounds,
+        GameActions,
+        NetworkFlush,
+        Scripts,
+    };
+
+    // ~6.5s at 40Hz
+    constexpr size_t LOGIC_UPDATE_MEASURMENTS_COUNT = 256;
+
+    // In order not to cause allocations, collect multiple samples into single pre-allocated struct
+    using LogicTimingInfo = std::map<LogicTimePart, std::array<std::chrono::duration<double>, LOGIC_UPDATE_MEASURMENTS_COUNT>>;
```

Comment:
```suggestion
    using LogicTimingInfo = std::unordered_map<LogicTimePart, std::array<std::chrono::duration<double>, LOGIC_UPDATE_MEASURMENTS_COUNT>>;
```
Doesn't seem like we really need the ordering